lib/fs/basicfs_copy_range_ioctl.go

```diff
@@ -56,38 +36,38 @@ func copyRangeIoctl(src, dst basicFile, srcOffset, dstOffset, size int64) error
 
 	if srcOffset == 0 && dstOffset == 0 && size == 0 {
 		// Optimization for whole file copies.
-		var errNo syscall.Errno
+		var innerErr error
 		_, err := withFileDescriptors(src, dst, func(srcFd, dstFd uintptr) (int, error) {
-			_, _, errNo = syscall.Syscall(syscall.SYS_IOCTL, dstFd, FICLONE, srcFd)
+			innerErr = unix.IoctlFileClone(int(dstFd), int(srcFd))
 			return 0, nil
 		})
 		// Failure in withFileDescriptors
 		if err != nil {
 			return err
 		}
-		if errNo != 0 {
-			return errNo
+		if innerErr != nil {
+			return innerErr
 		}
 		return nil
 	}
 
-	var errNo syscall.Errno
+	var innerErr error
 	_, err = withFileDescriptors(src, dst, func(srcFd, dstFd uintptr) (int, error) {
-		params := fileCloneRange{
-			srcFd:     int64(srcFd),
-			srcOffset: uint64(srcOffset),
-			srcLength: uint64(size),
-			dstOffset: uint64(dstOffset),
+		params := unix.FileCloneRange{
+			Src_fd:      int64(srcFd),
+			Src_offset:  uint64(srcOffset),
+			Src_length:  uint64(size),
+			Dest_offset: uint64(dstOffset),
 		}
-		_, _, errNo = syscall.Syscall(syscall.SYS_IOCTL, dstFd, FICLONERANGE, uintptr(unsafe.Pointer(&params)))
+		innerErr = unix.IoctlFileCloneRange(int(dstFd), &params)
 		return 0, nil
 	})
 	// Failure in withFileDescriptors
 	if err != nil {
 		return err
 	}
-	if errNo != 0 {
-		return errNo
+	if innerErr != nil {
+		return innerErr
```

Comment:
I think the innerErr magic can go now.

The reason is was there is because we used syscall.Errno which is always not nil, even if the errno is 0. 
We lose the sycall.Errno type as we return from withFileDescriptors, so we did the external variable magic to capture that.

The unix library now handles it for us, so we can just `return 0, errorFromUnixPackage` and then just return whatever was returned from withFileDescriptors straight back out.

Reply:
Done, thanks for the review.